src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -263,7 +263,7 @@ namespace cryptonote
     if(context.m_state == cryptonote_connection_context::state_synchronizing)
       return true;
 
-    if(m_core.have_block(hshd.top_id))
+    if(m_core.have_block(hshd.top_id) && !(hshd.current_height < m_core.get_target_blockchain_height()))
```

Comment:
I'd have simplified to && hshd.current_height >= m_core.get_target_blockchain_height() but doesn't really matter I guess. Maybe even reverse the two tests to avoid a pointless DB lookup :D